reference-implementation/lib/transform-stream.js

```diff
@@ -134,9 +139,15 @@ function TransformStreamTransform(transformStream, chunk) {
 
   transformStream._transforming = true;
 
+  const transformer = transformStream._transformer;
   const controller = transformStream._transformStreamController;
-  const transformPromise = PromiseInvokeOrNoop(transformStream._transformer,
-                             'transform', [chunk, controller]);
+  let transformPromise;
+  if (transformer.transform === undefined) {
```

Comment:
Oh, this is not great. We should only get the `transformer.transform` once. This gets it twice, once here and once inside PromiseInvokeOrNoop. I am not sure what a good design is.